js/build.js

```diff
@@ -22,18 +22,106 @@ Fliplet.Widget.instance('login-ds', function(data) {
+        passwordPlaceholder: 'Enter your password',
+        mismatch: 'The details you entered don’t match our records. Please try again.',
+        action: 'Log in',
+        loader: 'Loading...',
+        forgotPassword: 'Forgot your password?',
+      },
+      'component.login.restore': {
+        instruction: 'To reset your password you must verify your email address.',
+        mismatch: 'Your email address cannot be found. Please try again.',
+        goBack: 'Back to Login',
+        action: 'Verify email',
+        verifyCode: 'I have a verification code',
+      },
+      'component.login.reset': {
+        instruction: 'Enter your new password below.',
+        passwordPlaceholder: 'Enter your new password',
+        confirmPasswordPlaceholder: 'Confirm your new password',
+        mismatch: 'Passwords don\'t match. Please try again.',
+        action: 'Reset password',
+      },
+      'component.login.verify': {
+        instruction: 'A verification code was sent to',
+        codePlaceholder: 'Enter verification code',
+        mismatch: 'You entered the wrong code or the code expired. Please try again or request a new code.',
+        resend: 'Resend verification code',
+        action: 'Verify',
+        sendCode: 'A new code was sent to your email.',
+      },
+      'component.login.allDone': {
+        instruction: 'You can now login using your email address and new password.',
+        action: 'Login'
+      }
+    }
+  };
+  var currentLang = navigator.language;
 
   if (Fliplet.Navigate.query.error) {
     $container.find('.login-error').html(Fliplet.Navigate.query.error).removeClass('hidden');
   }
 
+  function initI18Next() {
+    i18next.init({
+      lng: currentLang,
+      fallbackLng: 'en',
+      resources: defaultLang 
+    }).then(function() {
+      setCurrentLang()
+
+      $container.css('visibility', 'visible');
+      // Fire hook for other features to adjust UI render if necessary
+      Fliplet.Hooks.run('i10nComplete');
+    });
+  }
+
+  function setCurrentLang() {
+    $('[data-i18n-key]').each(function() {
+      var $el = $(this);
+      var key = $el.data('i18n-key');
+  
+      if (!i18next.exists(key)) {
+        // i18n key not found
+        return;
+      }
+  
+      switch ($el.data('i18n-type')) {
+        case 'attr':
+          // Update HTML attribute
+          var attr = $el.data('i18n-attr');
+  
+          if (!attr) {
+            break;
+          }
+  
+          $el.attr(attr, i18next.t(key));
+          break;
+        default:
+          // Update inner HTML
+          $el.html(i18next.t(key));
+          break;
+      }
+    });
+  } 
+
   function initEmailValidation() {
     Fliplet.Navigator.onReady().then(function() {
       Fliplet.Security.Storage.init().then(function() {
         attachEventListeners();
         setUserDataPV(function() {}, function() {});
       });
 
+      Fliplet.Hooks.run('beforei18Init', { langs: defaultLang });
```

Comment:
1. Shouldn't you wait for this to finish before doing the init? It's a promise which you can wait for.
2. Name should be more generic like `beforeLocalize`

3. How can a use distinguish from this component and another component needing to fire the same hook? I have several doubts about this implementation, @tonytlwu we might need to discuss this because I don't want this being used without us getting involved in understanding risk and future plans.

e.g. we may want to use specific hook names such as `beforeLoginLocalize`

Reply:
1. I didn't know that it returns a promise and you are right using then, in this case, may save us from many errors in the future. I'll change this part of the code.
2. If you okay with `beforeLoginLocalize` hook name I would change it right away.

Reply:
@YaroslavOvdii yes, let's change the hook name as suggested